File: src/main/python/WordCount.py
import re
import sys
from pyspark import SparkContext

if __name__ == "__main__":
  if len(sys.argv) != 2:
    print "Usage: WordCount <file or directory>"
    exit(0)
  
  sc = SparkContext()
  pairs = (sc.textFile(sys.argv[1])
    .flatMap(lambda line: re.split("[. ;:,]+", line))
    .map(lambda word: (word,1))
    .filter(lambda (word, length): word)
    .reduceByKey(lambda v1,v2: v1+v2)
    .map(lambda (word,count): (count,word))
    .sortByKey(False)
    .map(lambda (count,word): (word,count)))
  for pair in pairs.take(5):
    print '{0}, {1}'.format(pair[0], pair[1])
  sc.stop()

File: src/main/scala/my/spark/examples/WordCount.scala
package my.spark.examples

import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.SparkConf

object WordCount {
  def main(args: Array[String]) {

    if (args.length != 1) {
      println("Usage: WordCount <file or directory>")
      System.exit(1)
    }

    val sc = new SparkContext()
    val pairs = (sc.textFile(args(0))
      .flatMap(line => line.split("\\.|\\,|\\;|\\s"))
      .map(word => (word,1))
      .filter(pair => pair._1.length > 0)
      .reduceByKey((v1,v2) => v1+v2)
      .map(pair => (pair._2, pair._1))
      .sortByKey(false)
      .map(pair => (pair._2, pair._1)))
    pairs.take(5).foreach(pair => {
      println(pair._1 + ", " + pair._2)
    })
    sc.stop()
  }
}
